types/google-apps-script/google-apps-script.url-fetch.d.ts

```diff
@@ -47,7 +48,7 @@ declare namespace GoogleAppsScript {
        * payload. It can be a string, a byte array, or a JavaScript object. A JavaScript object will be
        * interpretted as a map of form field names to values, where the values can be either strings or blobs.
        */
-      payload?: string;
+      payload?: Object;
```

Comment:
Lowercase `object`.
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

Reply:
I don't think this can be `object`.  The google apps script document says that:
>It can be a string, a byte array, or a JavaScript object.

However, `object` type isn't compatible to `string` because it is a type that represents the **non-primitive type**. 
https://www.typescriptlang.org/docs/handbook/basic-types.html#object

It means that if we use `object` instead of `Object`, the following code wouldn't pass type check:
```TypeScript
// test for URLFetchRequestOptions.payload
import URLFetchRequestOptions = GoogleAppsScript.URL_Fetch.URLFetchRequestOptions;
const postTest = (): string => {
  const url = 'http://httpbin.org/post';
  const params: URLFetchRequestOptions = {
    method: 'post',
    payload: JSON.stringify({test: 'test'})
  };
  return UrlFetchApp.fetch(url, params).getContentText();
};
```
Or, do I need to use `any`?

Reply:
Good point. Can you put:
`string | object | Byte[]`

It looks like the reference docs are incorrect.
https://developers.google.com/apps-script/reference/url-fetch/url-fetch-app

Reply:
Thanks. `Byte` is defined as `number` in `google-apps-script.types.d.ts`, but the following test code got an error `Attribute provided with invalid value: payload at [unknown function](bundle:134) at [unknown function](bundle:143)`:
```TypeScript
import URLFetchRequestOptions = GoogleAppsScript.URL_Fetch.URLFetchRequestOptions;
import Byte = GoogleAppsScript.Byte;

declare var global: any;

const postTest = (payload: {}): string => {
  const url = 'http://httpbin.org/post';
  const params: URLFetchRequestOptions = {
    method: 'post',
    payload: payload
  };
  return UrlFetchApp.fetch(url, params).getContentText();
};

global.main = (): void => {
  const bytes: Byte[] = [1, 2, 3];
  Logger.log(JSON.parse(postTest(bytes)).form);
};
```
I will try what types can be accepted tomorrow, but I guess it might be `Blob`.